packages/react-ui/src/components/molecules/MultipleSelectField.js

```diff
@@ -0,0 +1,250 @@
+          </Typography>
+        );
+      }
+      if (areAnySelected) {
+        return (
+          <Typography
+            component='span'
+            variant={isSmall ? 'body2' : 'body1'}
+            color='textPrimary'
+            ml={paddingSize}
+          >
+            {showCounter && currentOptions.length > 1
+              ? counterText
+              : currentOptions.join(', ')}
+          </Typography>
+        );
+      }
+      return (
+        <Typography
+          component='span'
+          variant={isSmall ? 'body2' : 'body1'}
+          color='textSecondary'
+          ml={paddingSize}
+        >
+          {placeholder || intlConfig.formatMessage({ id: 'c4r.form.noneSelected' })}
+        </Typography>
+      );
+    }, [
+      areAllSelected,
+      areAnySelected,
+      counterText,
+      currentOptions,
+      intlConfig,
+      isSmall,
+      paddingSize,
+      placeholder,
+      showCounter
+    ]);
+
+    return (
+      <SelectField
+        {...props}
+        ref={ref}
+        multiple
+        displayEmpty
+        placeholder={placeholder}
+        value={currentOptions}
+        renderValue={() => renderValue}
+        onChange={handleChange}
+        size={size}
+        labelSecondary={
+          showFilters ? (
+            <Filters>
+              <LinkFilter
+                variant='caption'
+                component='button'
+                disabled={areAllSelected || selectAllDisabled}
+                onClick={selectAll}
+              >
+                {intlConfig.formatMessage({ id: 'c4r.form.selectAll' })}
+              </LinkFilter>
+              <Typography variant='caption' weight='strong' color='text.hint'>
+                •
+              </Typography>
+              <LinkFilter
+                variant='caption'
+                component='button'
+                onClick={unselectAll}
+                disabled={!areAnySelected}
+              >
+                {intlConfig.formatMessage({
+                  id: 'c4r.form.selectNone'
+                })}
+              </LinkFilter>
+            </Filters>
+          ) : undefined
+        }
+      >
+        {options?.map((option) => {
+          const item = (
+            <StyledMenuItem
+              key={option.value}
+              value={option.value}
+              disabled={option.disabled}
+            >
+              <Checkbox
+                disabled={option.disabled}
+                checked={currentOptions.indexOf(option.value) > -1}
+              />
+              <ListItemText primary={option.label} />
+            </StyledMenuItem>
+          );
+          const content = option.tooltip ? (
+            <Tooltip
+              key={option.value}
+              title={option.tooltip}
+              placement={tooltipPlacement}
+            >
+              {item}
+            </Tooltip>
+          ) : (
+            item
+          );
+          return content;
+        })}
```

Comment:
We could consider extracting this into its own smaller component, to reduce a bit the size / complexity here

Reply:
I extracted the Filters and the logic into a hook, but this part needs more research, when I extract it into a component, it stops working... 

I prefer to do it in other PR, because I need to jump to the next task and I already did extra work here, like adding the filters themselves to be able to replace/remove code in cloud-native (which btw I also did and is not mandatory for the PFB).

So if you can approve the PR, I'll merge the branch and create the patch version 🙏

Reply:
Here are the changes: https://github.com/CartoDB/carto-react/pull/826/commits/c53155ba23fc3416d2a4eecc3fa28681b80f0fc7I extracted the Filters and the logic into a hook, but this part needs more research, when I extract it into a component, it stops working... 

I prefer to do it in other PR, because I need to jump to the next task and I already did extra work here, like adding the filters themselves to be able to replace/remove code in cloud-native (which btw I also did and is not mandatory for the PFB).

So if you can approve the PR, I'll merge the branch and create the patch version 🙏